### PR TITLE
#923 Let's implement RtRepo.languages().

### PR DESCRIPTION
**Pull request by nhekfqn**

### CODE REVIEW

**Comment by dmarkov** [comment]
Thanks for the pull request, let me find a reviewer..

**Comment by dmarkov** [comment]
@krzyk could you please review this pull request

File: src/main/java/com/jcabi/github/RtRepo.java [hunk 1]
```diff
@@ -46,15 +50,15 @@
...
- *  See https://developer.github.com/v3/repos/#list-languages for API details.
  */
 @Immutable
 @Loggable(Loggable.DEBUG)
 @EqualsAndHashCode(of = { "ghub", "entry", "coords" })
-@SuppressWarnings({ "PMD.TooManyMethods", "PMD.CouplingBetweenObjects" })
+@SuppressWarnings({
+        "PMD.TooManyMethods",
+        "PMD.CouplingBetweenObjects",
+        "PMD.AvoidInstantiatingObjectsInLoops"
```
Comment (krzyk) [inline src/main/java/com/jcabi/github/RtRepo.java]: It is better to have suppress closer to the source, here `PMD.AvoidInstantiatingObjectsInLoops` should be for `languages()` method

File: src/main/java/com/jcabi/github/RtRepo.java [hunk 2]
```diff
@@ -216,7 +220,27 @@ public Notifications notifications() {
 
     @Override
     public Iterable<Language> languages() {
-        return Collections.emptyList();
+        final RtJson rtJson = new RtJson(
+                this.request.uri()
+                .path("/languages")
+                .back()
+        );
+        JsonObject json;
```
Comment (krzyk) [inline src/main/java/com/jcabi/github/RtRepo.java]: json can be final

File: src/main/java/com/jcabi/github/RtRepo.java [hunk 3]
```diff
@@ -216,7 +220,27 @@ public Notifications notifications() {
...
+        try {
+            json = rtJson.fetch();
+        } catch (final IOException ex) {
+            throw new IllegalStateException(ex);
+        }
+        final int size = json.size();
```
Comment (krzyk) [inline src/main/java/com/jcabi/github/RtRepo.java]: `size` can be inlined

File: src/main/java/com/jcabi/github/RtRepo.java [hunk 4]
```diff
@@ -216,7 +220,27 @@ public Notifications notifications() {
...
+        final List<Language> languages = new ArrayList<Language>(size);
+        for (final Map.Entry<String, JsonValue> value : json.entrySet()) {
+            final String name = value.getKey();
+            final JsonNumber bytesJsonNumber = json.getJsonNumber(name);
+            final long bytes = bytesJsonNumber.longValue();
+            final Language language = new RtLanguage(name, bytes);
```
Comment (krzyk) [inline src/main/java/com/jcabi/github/RtRepo.java]: `language` can be inlined

File: src/main/java/com/jcabi/github/RtRepo.java [hunk 5]
```diff
@@ -216,7 +220,27 @@ public Notifications notifications() {
...
+        try {
+            json = rtJson.fetch();
+        } catch (final IOException ex) {
+            throw new IllegalStateException(ex);
+        }
+        final int size = json.size();
+        final List<Language> languages = new ArrayList<Language>(size);
+        for (final Map.Entry<String, JsonValue> value : json.entrySet()) {
+            final String name = value.getKey();
+            final JsonNumber bytesJsonNumber = json.getJsonNumber(name);
```
Comment (krzyk) [inline src/main/java/com/jcabi/github/RtRepo.java]: `bytesJsonNumber` can be inlined

File: src/main/java/com/jcabi/github/RtRepo.java [hunk 6]
```diff
@@ -216,7 +220,27 @@ public Notifications notifications() {
...
+            final long bytes = bytesJsonNumber.longValue();
```
Comment (krzyk) [inline src/main/java/com/jcabi/github/RtRepo.java]: `bytes` can be inlined

File: src/test/java/com/jcabi/github/RtRepoTest.java [hunk 7]
```diff
@@ -303,11 +302,21 @@ public void fetchNotifications() {
 
     /**
      * RtRepo can fetch languages.
+     *
```
Comment (krzyk) [inline src/test/java/com/jcabi/github/RtRepoTest.java]: empty space after method description is redundant

File: src/test/java/com/jcabi/github/RtRepoTest.java [hunk 8]
```diff
@@ -303,11 +302,21 @@ public void fetchNotifications() {
...
+            new MkAnswer.Simple(
+                HttpURLConnection.HTTP_OK,
+                Json.createObjectBuilder()
+                    .add("Ruby", 1)
+                    .build().toString()
+            )
+        ).start();
         final Repo repo = RtRepoTest.repo(
-            new FakeRequest()
+                new ApacheRequest(container.home())
```
Comment (krzyk) [inline src/test/java/com/jcabi/github/RtRepoTest.java]: I think indentation here is off

**Comment by krzyk** [comment]
@nhekfqn couple of comments above

**Comment by nhekfqn** [comment]
@krzyk defects fixed, please take a look.

@yegor256 There is a merge conflict in class comments to RtRepo

<<<<<<< HEAD
=======
 * @todo #923 Let's implement RtRepo.languages().
 *  Now it returns empty iterable but should return list of repo languages.
 *  Uncomment test RtRepoTest.iteratesLanguages() when done.
 *  See https://developer.github.com/v3/repos/#list-languages for API details.
 * @todo #955 Uncomment test RtRepoTest.iteratesLanguages()
 *  when RtRepoITCase.languages() implemented.
>>>>>>> bd8d95bb204f7ad634c244a3eee1d0f069e6517b

is ok for person who will merge changes to resolve it (remove todo #923) or should I merge and resend pull request?

**Comment by krzyk** [comment]
@nhekfqn you should merge the changes from master

**Comment by nhekfqn** [comment]
Replaced with https://github.com/jcabi/jcabi-github/pull/978

**Comment by dmarkov** [comment]
@krzyk thank you, added **16 mins** to your acc, payment num is `50258010`. +16 added to your rating, at the moment it is: [+1429](http://www.netbout.com/b/35035?open=rating)

**Pull request closed without merging**